docs/lib/storage/fragments/js/upload.md

```diff
@@ -95,17 +95,93 @@ async function onChange(e) {
 Upload an image in your React Native app:
 
 ```javascript
-async function pathToImageFile() {
-  try {
-    const response = await fetch(pathToImageFile);
-    const blob = await response.blob();
-    await Storage.put('yourKeyHere', blob, {
-      contentType: 'image/jpeg', // contentType is optional
+async function pathToImageFile(data) {
+      try {
+        const response = await fetch(data);
+        const blob = await response.blob();
+        await Storage.put(`your key here`, blob, {
+          contentType: 'image/jpeg', // contentType is optional
+        });
+      } catch (err) {
+        console.log('Error uploading file:', err);
+      }
+    }
+    pathToImageFile(result.uri);
+  }
+```
+
+
+## React Native upload images with Expo Image Picker
+Expo Image Picker is a commonly used image library for React Native. Combined with S3 it is a powerful tool for any workflow. Make sure you set up your Auth and Storage, then copy the code below:
+
+```javascript
+import { withAuthenticator } from 'aws-amplify-react-native'
+import Amplify, { Storage } from 'aws-amplify'
+import config from './src/aws-exports'
+// import awsconfig from './aws-exports';
+// Might need to switch line 7 to awsconfig 
+Amplify.configure(config)
+
+import { StatusBar } from 'expo-status-bar';
+import React, { useState, useEffect } from 'react';
+import { Button, Image, View, Platform, StyleSheet, Text, TextInput } from 'react-native';
+import * as ImagePicker from 'expo-image-picker';
+
+function App() {
+  const [image, setImage] = useState(null)
+  const [name, setName] = useState('Evan Erickson')
```

Comment:
This feels superfluous without any usage of `setName` in the example, can `name` just be added as an argument to `pathToImageFile`, e.g. 
```
async function pathToImageFile(data, name) { ... }
```